### PR TITLE
Integration n+1

### PR DESCRIPTION
**Pull request by tsohleDev**
# Fix n+1 problem, Integration spec

In this PR we:
- Fix the n+q problem with the include #
- Use Capybara to write integration tests

| before | after |
| --- | --- |
| PR pic | PR pic |

### CODE REVIEW

File: spec/views/posts/index.spec [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  end
+  it 'can see how many comments a post has' do
+    visit "/users/#{@user.id}/posts"
+    expect(page).to have_text('Comments:')
+  end
+  it 'can see how many likes a post has' do
+    visit "/users/#{@user.id}/posts"
+    expect(page).to have_text('Likes:')
+  end
+end
```
Comment (rotimiazeez) [inline spec/views/posts/index.spec]: - Please ensure the below tests are added to your file to complete the requirement for this milestone 😉
```
Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
Tests that redirects to the post's show page when you click on a post.
```

File: spec/views/user/index.spec [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    end
+  end
+
+  it 'displays images of all users' do
+    visit '/users'
+    @users.each do |user|
+      expect(page).to have_selector("img[src='#{user.photo}']")
+    end
+  end
+end
```
Comment (rotimiazeez) [inline spec/views/user/index.spec]: - Please ensure the below tests are added to your file to complete the requirement for this milestone 😉
```
Tests that asserts you can see the number of posts each user has written.
Tests that redirects to the user's show page when you click on a user.
```

File: spec/views/user/show.spec [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      expect(page).to have_text('Bio')
+      expect(page).to have_text(user.bio.to_s)
+    end
+  end
+
+  it 'can see a button that lets me view all of a user posts' do
+    visit "/users/#{first_user.id}"
+    expect(page).to have_text('See all posts')
+  end
+end
```
Comment (rotimiazeez) [inline spec/views/user/show.spec:93]: - Please note that the below are what is suppose to be tested for `user_show_spec` to complete the requirement for this milestone 😉
```
Tests that asserts you can see the post's title.
Tests that asserts you can see who wrote the post.
Tests that asserts you can see how many comments it has.
Tests that asserts you can see how many likes it has.
Tests that asserts you can see the post body.
Tests that asserts you can see the username of each commentor.
Tests that asserts you can see the comment each commentor left.
```

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi TEAM 👍🏽,

## Good job so far!
### To Highlight 👇🏽
- Good teamwork 👏🏽
- Tests implemented 😉
- No Linter error ✔️ 
- Good PR description 👍🏽 
- Good commit message(s) 😉
- Descriptive README 📖

You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

File: spec/views/user/show.spec (hunk 3 above)
Comment (tsohleDev) [reply spec/views/user/show.spec:93]: I don't think that the comments should be viewed by the user

File: spec/views/user/show.spec [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        expect(page).to have_text("Likes: #{post.title}") if post.likes_counter > 0
+      end
+    end
+  end
+
+  it 'can see a button that lets me view all of a user posts' do
+    visit "/users/#{first_user.id}"
+    expect(page).to have_text('See all posts')
+  end
+end
```
Comment (franclobo) [inline spec/views/user/show.spec:93]: Hello @tsohleDev!

We've just realized that the requested change corresponds to the post#show view tests and not to the user#show view tests.
We are sorry to mislead you. We haven't done that on purpose 😊 

- **[OPTIONAL]** Please 🙏🏽 complete the tests on the user#show view, kindly add the following tests 👇🏽 

> Tests that asserts you can see the user's first 3 posts.
> Tests that asserts you can see a button that lets me view all of a user's posts.
> Tests that redirects to the post's show page when you click on a user's post.
> Tests that redirects to the user's post's index page when you click on a see all posts.

**Review by franclobo — APPROVED** [review]
Hi @tsohleDev, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Correct Git flow ✔️ 
- Descriptive PR ✔️ 
- All tests passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**